src/app/ReadClient.h

```diff
@@ -234,6 +234,22 @@ class ReadClient : public Messaging::ExchangeDelegate
             aEventNumber.ClearValue();
             return CHIP_NO_ERROR;
         }
+
+        /**
+         * OnUnsolicitedCommunication will be called for a subscription ReadClient
+         * when any incoming communication happens with a matching node on
+         * the fabric.
+         *
+         * This callback will be called:
+         *   - When receiving any unsolicited communication from the node
+         *   - Even for disconnected subscriptions.
+         *
+         * @param[in] apReadClient the ReadClient for the subscription.
+         */
+        virtual void OnUnsolicitedCommunication(ReadClient * apReadClient)
```

Comment:
Per the docs here (which rever to "communication" in general), what makes this communication "unsolicited"?  Should this be named something more like "OnMessageReceivedFromPublisher" or something?

Will Read-type clients ever call this?  Needs to be documented.

The current implementation does in fact only call this for unsolicited data reports, but it seems like as a followup we could push this down so any CASE-authenticated message received from the publisher could trigger this.....

Reply:
I can rename to "OnMessageReceivedFromPublisher" - this is a good way to describe exactly what we want to express here, in order to help applications talk to IC devices. I will rename these.

I think giving a callback for any received message is out of scope for this PR.

Reply:
Actually, perhaps "OnUnsolicitedMessageFromPublisher"?

Reply:
`OnUnsolicitedMessageFromPublisher` seems good.